packages/python/plotly/plotly/express/_core.py

```diff
@@ -1438,42 +1575,6 @@ def build_dataframe(args, constructor):
         value_name = _escape_col_name(columns, "value", [])
         var_name = _escape_col_name(columns, var_name, [])
 
-    if needs_interchanging:
```

Comment:
All this block is addressed above together with parsing/converting the input into narwhals object